app/services/FileUploadAnswersService.scala

```diff
@@ -17,35 +17,34 @@
 package services
 
 import models.FileUploadAnswers
-import org.joda.time.{DateTime, DateTimeZone}
 import play.api.Logging
-import play.api.libs.json.{JsObject, Json}
 import repositories.FileUploadAnswersRepository
 
+import java.time.{ZoneOffset, ZonedDateTime}
 import javax.inject.Inject
 import scala.concurrent.{ExecutionContext, Future}
 
 class FileUploadAnswersService @Inject()(val repository: FileUploadAnswersRepository)(implicit ec: ExecutionContext) extends Logging {
 
-  def removeByEori(eori: String): Future[Unit] = repository.remove("eori" -> eori).map(_ => (): Unit)
+  def findOne(eori: String): Future[Option[FileUploadAnswers]] = repository.findOne(eori)
 
-  def findByEori(eori: String): Future[Option[FileUploadAnswers]] = repository.find("eori" -> eori).map(_.headOption)
+  def findOneAndRemove(eori: String): Future[Option[FileUploadAnswers]] = repository.findOneAndRemove(eori)
 
-  def findOrCreate(eori: String): Future[FileUploadAnswers] =
-    findByEori(eori).flatMap {
-      case Some(movementCache) => Future.successful(movementCache)
-      case None                => save(FileUploadAnswers(eori))
+  def findOneOrCreate(eori: String): Future[FileUploadAnswers] =
+    repository.findOneOrCreate(eori) map {
+      case Some(answers) => answers
+      case None =>
+        logger.warn(s"Error when persisting file upload answers for eori($eori)")
+        FileUploadAnswers(eori)
```

Comment:
Not sure when `findOneOrCreate` would return a successful Future with a None present (if no existing record is found and it fails to create a new record I'd expect the Future to fail). But if I'm reading this correctly, in this odd situation we return a new record that has not actually been persisted (which seems wrong) :thinking: